### PR TITLE
[RP Workflows] Unify the PreprocessShaders code for URP and HDRP

### PR DESCRIPTION
**Pull request by alex-vazquez**
### Purpose of this PR
- Improvements and code parity between HDRP and URP for `IPreprocessShaders`
- URP is now exporting the shader stripping as HDRP was doing on the file `Temp/shader-strip.json`
- Shader export file only dumps the one single line per shader and pass.
- Fixing LogOnlyURP shaders.
- Enabling Profiling HDRP stripping as URP was doing with define `PROFILE_BUILD`
- Code clean up and refactor for common parts on core package.

---
### Testing status
Make sure shader stripping is stripping the same amount of variants as was being done before.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/BuildProcessors/HDRPPreprocessShaders.cs [hunk 1]
```diff
@@ -419,160 +414,101 @@ public HDRPreprocessShaders()
...
+            {
+                case ShaderVariantLogLevel.Disabled:
+                    return false;
+                case ShaderVariantLogLevel.OnlyHDRPShaders:
+                    return HDShaderUtils.IsHDRPShader(shader);
+                case ShaderVariantLogLevel.AllShaders:
+                    return true;
             }
+
+            throw new System.Exception("Missing ShaderVariant Log Level");
```
[the remark this answers is not in the document]
Comment (alex-vazquez) [reply com.unity.render-pipelines.high-definition/Editor/BuildProcessors/HDRPPreprocessShaders.cs:436]: I think is a dev error that needs to be adressed if a Variant is added, so in my opinion is accepted.

[the remark this answers is not in the document]
Comment (fredericv-unity3d) [reply com.unity.render-pipelines.high-definition/Editor/BuildProcessors/HDRPPreprocessShaders.cs:436]: Let me clarify: I agree this error is to be addressed. But do we expect to throw an exception? Is there a try/catch statement in the call chain to catch this?

Or should we return an error instead, so the caller is aware of the error without using the exception flow?

[the remark this answers is not in the document]
Comment (alex-vazquez) [reply com.unity.render-pipelines.high-definition/Editor/BuildProcessors/HDRPPreprocessShaders.cs:436]: Ok I see your point. I will change it directly to an Log Error

**Review by sebastienlagarde — APPROVED** [review]
diff on HDRP part is a bit hard to read, but guess it is fine.

**Pull request closed without merging**